### PR TITLE
fix y-axis scatter extend calculation

### LINKED ISSUE #20822: incorrect positioning for log scale of y-axis in scatter plot
**Describe the bug**
When displaying a scatter plot, changing the y-axis to log scale results in incorrect positioning of data points. No problem with x-axis log scale.

**To Reproduce**

1. Add the attached sqlite3 database: [satellites.zip](https://github.com/metabase/metabase/files/8177583/satellites.zip)
2. Browse to the sat table in the satellites database.
3. Choose "visualization" => "scatter"
4. Select "orbit major" as x-axis and "rev cycle" as y-axis.
5. On the axes tab, choose log scale for y-axis

**Expected behavior**

These are orbit parameters of natural satellites in the solar system. According to [Kepler's 3rd law](https://en.wikipedia.org/wiki/Kepler%27s_laws_of_planetary_motion#Third_law), each group of satellites revolving around the same parent planet should exhibit the relation rev_cycle^2 ~ orbit_major^3. In a plot with both axes in log scale, all satellites around the same planet should lie on a straight line. This pattern can be observed in [my log scale drawing](https://ckhung.github.io/scatplot/) using the same set of data in csv format.

**Screenshots**

In metabase, the picture seems correct when both axes are linear scale, or when x axis is log scale and y axis is linear scale. It does not look right when y-axis is switched to log scale. Here are the screenshots of the four combinations of x-y: linear-linear, log-linear, linear-log, and log-log.
- Correct, X=Linear and Y=Linear
lin-lin
- Correct, X=Log and Y=Linear
log-lin
- Incorrect, X=Linear and Y=Log
lin-log
- Incorrect, X=Log and Y=Log
log-log

**Information about my Metabase Installation:**

```
{
  "browser-info": {
    "language": "en",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:94.0) Gecko/20100101 Firefox/94.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.11+9-Ubuntu-0ubuntu2.20.04",
    "java.vendor": "Ubuntu",
    "java.vendor.url": "https://ubuntu.com/",
    "java.version": "11.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.11+9-Ubuntu-0ubuntu2.20.04",
    "os.name": "Linux",
    "os.version": "5.8.0-59-generic",
    "user.language": "zh",
    "user.timezone": "Asia/Taipei"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "sqlite"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-02-28",
      "tag": "v0.42.2",
      "branch": "release-x.42.x",
      "hash": "d6ff494"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

- Browser: Firefox 94.0
- OS: Linux mint 20.2
- DB: sqlite3 3.31.1-4ubuntu0.2
- Metabase version:  0.42.2
- Metabase hosting environment: localhost
- Metabase internal database: H2

**Severity**

mild

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/20822

## Changes

While grouping data for scatter chart it used wrong index for y-axis values. This led to invalid y-axis extent calculation [1, 1] which somehow worked with a linear scale correctly.

## How to verify

- Create a native query
```
select 1 x, 1 y
union all  select 10 x, 10 y
union all  select 100 x, 100 y
union all  select 200 x, 200 y
union all  select 10000 x, 10000 y
```
- Change y-axis type to log and ensure it looks correct

**Before**
Screenshot 2022-04-28 at 03 09 53

**After**
Screenshot 2022-04-28 at 03 09 25

### CODE REVIEW

**Comment by alxnddr** [comment]
@metabase-bot run visual-tests

File: frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js [hunk 1]
```diff
@@ -126,13 +126,11 @@ function getXAxisProps(props, datas, warn) {
 ///------------------------------------------------------------ DIMENSIONS & GROUPS ------------------------------------------------------------///
 
 function getDimensionsAndGroupsForScatterChart(datas) {
-  const dataset = crossfilter();
-  datas.map(data => dataset.add(data));
-
+  const dataset = crossfilter(datas);
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js:129]: This is unrelated but it [did not seem to make sense not to pass](https://github.com/crossfilter/crossfilter/wiki/API-Reference) `datas` while creating crossfilter dataset

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

**Review by ranquild — APPROVED** [review]
Could you add a repro for this issue? Thanks!

**Review by iethree — APPROVED** [review]

**Pull request merged**